sdcm/audit.py

```diff
@@ -0,0 +1,229 @@
+    date: date
+    node: str
+    event_time: datetime
+    category: AuditCategory
+    consistency: str
+    error: bool
+    keyspace_name: str
+    operation: str
+    source: str
+    table_name: str
+    username: str
+
+
+class AuditLogReader:  # pylint: disable=too-few-public-methods
+
+    def __init__(self, cluster):
+        self._cluster = cluster
+
+    def read(self, from_datetime: Optional[datetime] = None,
+             category: Optional[AuditCategory] = None,
+             operation: Optional[str] = None,
+             limit_rows: int = 1000
+             ) -> List[AuditLogRow]:
+        raise NotImplementedError()
+
+
+class TableAuditLogReader(AuditLogReader):  # pylint: disable=too-few-public-methods
+
+    def read(self, from_datetime: Optional[datetime] = None,
+             category: Optional[AuditCategory] = None,
+             operation: Optional[str] = None,
+             limit_rows: int = 1000
+             ) -> List:
+        """Return audit log rows based on the given filters."""
+        where_list = []
+        if from_datetime:
+            where_list.append(f"event_time > {uuid_from_time(from_datetime.timestamp())}")
+        if category:
+            where_list.append(f"category = '{category}'")
+        if operation:
+            where_list.append(f"operation = '{operation}'")
+
+        limit = f" limit {limit_rows}" if limit_rows else ""
+        where = " where " + " and ".join(where_list) + f"{limit} ALLOW FILTERING" if where_list else ""
+        LOGGER.debug("Audit query filter: %s", where)
+        with self._cluster.cql_connection_patient(node=self._cluster.nodes[0]) as session:
+            rows = list(session.execute(f"select * from audit.audit_log{where} using timeout 5m"))
+        rows = [AuditLogRow(date=row.date.date(),
+                            node=row.node,
+                            event_time=datetime_from_uuid1(row.event_time),
+                            category=row.category,
+                            consistency=row.consistency,
+                            error=row.error,
+                            keyspace_name=row.keyspace_name,
+                            operation=row.operation,
+                            source=row.source,
+                            table_name=row.table_name,
+                            username=row.username) for row in rows
+                ]
+        LOGGER.debug("Found %s audit log rows", len(rows))
+        return rows
+
+
+def get_audit_log_rows(node,  # pylint: disable=too-many-locals
+                       from_datetime: Optional[datetime] = None,
+                       category: Optional[AuditCategory] = None,
+                       operation: Optional[str] = None,
+                       limit_rows: int = 1000
+                       ) -> List[AuditLogRow]:
+    with node.open_system_log(on_datetime=from_datetime) as log_file:
+        found_rows = 0
+        for line in log_file:
+            if '!NOTICE | scylla-audit' in line:
```

Comment:
I would be careful to depend on the log line structure that much, it might change, for example on VMs vs on docker/k8s or artifacts test when we use ssh for logs.

I would look for the level  "NOTICE" and for the source/component: "scylla-audit"

Reply:
Anyhow this type of code kind of calls for a unittest... (reading logs is always behaves a bit differently then we anticipate)

Reply:
this code part might explian why it was seem to not work in: 
https://github.com/scylladb/scylla-operator/issues/1299
from my POV, we can set it to not work at all in k8s for now.

I would recommend testing first with docker backend, and only then with k8s. (anyhow both are nice to have, and not a priority for now)

Reply:
improved and added unit tests